seeker/log/journald.go

```diff
@@ -0,0 +1,126 @@
+package log
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/hcdiag/seeker"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+var _ seeker.Runner = Journald{}
+
+// JournaldTimeLayout custom go time layouts must match the reference time Jan 2 15:04:05 2006 MST
+const JournaldTimeLayout = "2006-01-02 15:04:05"
+
+type Journald struct {
+	service string
+	destDir string
+	since   time.Time
+	until   time.Time
+}
+
+// NewJournald sets the defaults for the journald runner and returns a seeker
+func NewJournald(service, destDir string, since, until time.Time) *seeker.Seeker {
+	return &seeker.Seeker{
+		Identifier: "journald",
+		Runner: Journald{
+			service: service,
+			destDir: destDir,
+			since:   since,
+			until:   until,
+		},
+	}
+}
+
+// Run attempts to pull logs from journald via shell command, e.g.:
+// journalctl -x -u {name} --since '3 days ago' --no-pager > {destDir}/journald-{name}.log
+func (j Journald) Run() (interface{}, seeker.Status, error) {
+	// Check if systemd has a unit with the provided name, return a nil pointer
```

Comment:
Nitpick: To me, the wording on this comment is just a little bit unclear. I made a suggestion that I think clarifies it a bit.

```suggestion
	// Check if systemd has a unit with the provided name, and return a nil pointer if not.
```

Reply:
The "return a nil pointer" behavior this refers to is no longer valid so i'll just delete that section of the comment